sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQuerySpec.scala

```diff
@@ -89,6 +89,41 @@ class SparkE2EQuerySpec extends IntegrationSpec with Spark with CosmosClient wit
         rowWithInference.getAs[Boolean]("isAlive") shouldEqual true
     }
 
+    "spark query" can "use schema inference with custom query" taggedAs (RequiresCosmosEndpoint) in {
+        val cosmosEndpoint = TestConfigurations.HOST
+        val cosmosMasterKey = TestConfigurations.MASTER_KEY
+
+        val container = cosmosClient.getDatabase(cosmosDatabase).getContainer(cosmosContainer)
+        for (state <- Array(true, false)) {
+            val objectNode = Utils.getSimpleObjectMapper.createObjectNode()
+            objectNode.put("name", "Shrodigner's duck")
```

Comment:
did Shrodigner have a duck or cat? :-)
kidding :p
https://en.wikipedia.org/wiki/Schr%C3%B6dinger%27s_cat#:~:text=In%20quantum%20mechanics%2C%20Schr%C3%B6dinger's%20cat,may%20or%20may%20not%20occur.

Reply:
Hahaha I picked another animal because all the tests in this Spec run on the same container created by the trait, and it would pick the cat created in the first test, so I wanted something unique for this one, and I like ducks 😄

Reply:
there is builtin support for auto-cleaning container after each test

you can use the existing `AutoCleanableCosmosContainer` construct for test:

you can change
```scala
class SparkE2EQuerySpec extends IntegrationSpec with Spark with CosmosClient with CosmosContainer
```

to 
```scala
class SparkE2EQuerySpec extends IntegrationSpec with Spark with CosmosClient with AutoCleanableCosmosContainer
```

anyway ducks are good too :p

Reply:
Good learning on the `AutoCleanableCosmosContainer` trait, thanks for sharing!!